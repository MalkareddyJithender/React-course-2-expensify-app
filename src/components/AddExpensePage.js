import React from 'react';
import {connect} from 'react-redux';
import {startAddExpense,addExpense} from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

//to avoid inline functions we are creating class based component

export class AddExpensePage extends React.Component
{
    onSubmit = (expense) => 
    {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
    };

    render()
    {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm 
                onSubmit={this.onSubmit}
                />
            </div>
        );
    };
}


const mapDispatchToProps = (dispatch) =>({ startAddExpense:(expense) => dispatch(startAddExpense(expense))});

export default connect(undefined,mapDispatchToProps)(AddExpensePage);