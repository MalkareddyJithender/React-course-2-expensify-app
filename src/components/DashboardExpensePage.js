import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const DashboardExpensePage = () =>  (
    <div>
       <ExpenseListFilters />
       <ExpenseList />
    </div>
);


export default DashboardExpensePage;