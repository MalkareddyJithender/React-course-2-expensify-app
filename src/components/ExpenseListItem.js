import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import currencyFormatter from 'currency-formatter';



const ExpenseListItem = ({id,description,amount,createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
        </Link>
        <p>
            {currencyFormatter.format(amount/100,{code:'INR'})}            
              -     
            {moment(createdAt).format('MMMM Do, YYYY')}
             </p>
    </div>
);

export default ExpenseListItem;