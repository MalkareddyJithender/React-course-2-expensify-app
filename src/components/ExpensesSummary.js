import React from 'react';
import getVisibleExpenses from '../selecters/expenses';
import getExpensesTotal from '../selecters/expenses-total';
import {connect} from 'react-redux';
import currencyFormatter from 'currency-formatter';


export class ExpensesSummary extends React.Component
{
    render()
    {
        const expenseWord = this.props.expensesCount === 1 ? 'expense' : 'expenses';
        return (
            <div>
                {
                <h4> viewing {this.props.expensesCount} {expenseWord} totalling 
                { currencyFormatter.format(this.props.expensesTotal/100,{code:'INR'})} 
                </h4> 
                }
            </div>
        );
    }
};

const mapStateToProps = (state) =>
{
    const visibleExpenses = getVisibleExpenses(state.expenses,state.filters);
    return {
        expensesCount:visibleExpenses.length,
        expensesTotal:getExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);