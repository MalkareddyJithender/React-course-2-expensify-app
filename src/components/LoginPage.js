import React from 'react';
import { startLogin } from '../actions/auth';
import { connect } from 'react-redux';

export class LoginPage extends React.Component
{
    render()
    {
        return (
            <div>
                <button onClick={this.props.startLogin} >Login</button>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) =>({
    startLogin:() => dispatch(startLogin())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);