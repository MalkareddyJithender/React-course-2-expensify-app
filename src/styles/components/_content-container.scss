.content-container
{
    margin:0 auto;
    max-width:80rem;
    padding:0 $m-size;
}

.content-container--profile
{
    @extend .content-container;
    max-width:90rem;
    margin-top:$s-size;
    padding:$s-size 0;
    border:1px solid lighten($grey,50);

    @media(max-width:$desktop-breakpoint - 1)
    {
        margin:$s-size $s-size 0 $s-size;
    }
}

.content-container--About
{
    @extend .content-container;
    max-width:90rem;
    padding:0;
    margin-top:$m-size;
    border:1px solid lighten($grey,50);

    @media(max-width:$desktop-breakpoint - 1)
    {
        margin:$s-size $s-size $s-size $s-size;
    }
}