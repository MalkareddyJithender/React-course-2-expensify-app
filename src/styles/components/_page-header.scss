.page-header
{
    background:$off-white;
    padding:$m-size 0;
    margin-bottom:$l-size;
}

.page-header__title
{
    margin:0;
    font-weight:300;
    span{
        font-weight:700;
    }
}

.page-header__subtitle
{
    font-weight:300;
    font-family:sans-serif;
    span{
        font-weight:700;
    }
    margin:$s-size;
}

.page-header__actions
{
    margin-top:$m-size;
}

.page-header__user
{
    display:flex;
    justify-content: space-between;
    align-items:center;
}

.page-header__user__name
{
    margin:0;
    font-weight:300;
    font-size:$font-size-large;
    font-family:verdana;
}

.page-header__user__profile
{
    text-decoration:none;
    color:green;
    margin-right:$l-size;
}

#name
{
    color:green;
    font-weight:400;
}