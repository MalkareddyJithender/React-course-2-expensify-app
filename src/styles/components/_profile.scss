.img
{
    height:15rem;
    width:15rem;
    border:none;
    border-radius:100%;
}

.i-icon
{
    height:1.3rem;
    width:1.3rem;
    border-radius: 100%;
    margin-left:8px;
    margin-bottom:2px;
}

#profile
{
    display:flex;
    justify-content:space-evenly;
    flex-direction:row;

    @media(max-width:$desktop-breakpoint)
    {
        flex-direction:column;
        justify-content:center;
        align-items:center;
    }
}

.title
{
    margin:4.5rem 0 0 0;
    font-weight:300;
    font-family:sans-serif;
    color:rgb(165, 165, 165);

    @media(max-width:$desktop-breakpoint - 1)
    {
        margin-top:1rem;
    }
}

.About__title
{
    display: flex;
    align-items: center;
    background:$blue;
    color:white;
    font-weight:300;
    border-radius:1px;
}

.About__title__2
{
    margin:0;
}

.About__content
{
    margin:$m-size;
    font-weight:300;
}

.About__content__header
{
    margin:0;
    font-weight:100;
    font-size:1.5rem;
    color:ligthen($grey,100%);
}

.About__content__body
{
    margin:0;
    font-weight:700;
    font-size:$font-size-small;
    border-bottom:1px solid lighten($grey,50%);
}
