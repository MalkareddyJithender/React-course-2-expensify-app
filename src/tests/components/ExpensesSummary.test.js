import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render Expenses Summary component correctly with 1 expense',() =>
{
    const result = shallow(<ExpensesSummary 
        expensesCount={1}
        expensesTotal={1000000}
        hiddenExpensesCount={0}
        userName="Malkareddy Jithender"
           />);
    expect(result).toMatchSnapshot();
});

test('should render Expenses Summary component correctly with multiple expenses',() =>
{
    const result = shallow(<ExpensesSummary 
        expensesCount={5} 
        expensesTotal={23445000} 
        hiddenExpenses={2}
        userName="Malkareddy Jithender" 
        />);
    expect(result).toMatchSnapshot();
});