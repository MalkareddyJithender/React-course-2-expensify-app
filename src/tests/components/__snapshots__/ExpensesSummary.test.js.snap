// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Summary component correctly with 1 expense 1`] = `
<div>
  <h4>
     viewing 
    1
     
    expense
     totalling
    ₹10,000.00
  </h4>
</div>
`;

exports[`should render Expenses Summary component correctly with multiple expenses 1`] = `
<div>
  <h4>
     viewing 
    5
     
    expenses
     totalling
    ₹234,450.00
  </h4>
</div>
`;
